src/replication.c

```diff
@@ -2206,6 +2200,14 @@ void readSyncBulkPayload(connection *conn) {
             dbarray = diskless_load_tempDb;
             functions_lib_ctx = temp_functions_lib_ctx;
         } else {
+            /* We will soon start loading the RDB from socket, the replication history is changed,
+             * we must discard the cached primary structure and force resync of sub-replicas. */
+            replicationAttachToNewPrimary();
+
+            /* Even though we are on-empty-db and the database is empty, we still call emptyData. */
+            serverLog(LL_NOTICE, "PRIMARY <-> REPLICA sync: Flushing old data");
+            emptyData(-1, empty_db_flags, replicationEmptyDbCallback);
```

Comment:
Is there any benefit move this part after connSetReadHandler(conn, NULL);

Reply:
The benefit is that if the full sync fails, the replica didn't delete the old keys on the node.

Reply:
the code path here is in diskless load, the position does not matter? since remove the read handler won't affect anything here.

The real changes is in the later disk-based part. line 2333

Reply:
> The benefit is that if the full sync fails, the replica didn't delete the old keys on the node.

No, the changed line 2203 is in the diskless load part, it is not disk-based part.  The running condition is if use_diskless_load is true and server.repl_diskless_load != REPL_DISKLESS_LOAD_SWAPDB.  So I think the part code changes are not related to this PR, it confused me.

Reply:
> the code path here is in diskless load, the position does not matter? since remove the read handler won't affect anything here.
> 
> The real changes is in the later disk-based part. line 2333

yes, I agree with this part.

Reply:
in the old code, on-empty-db and disk-based will need to flush the db, and i want to move the position of the code, so in here, the on-empty-db mode also have the changes